cluster/images/etcd/migrate-if-needed.sh

```diff
@@ -46,7 +46,8 @@ set -o nounset
 BUNDLED_VERSIONS="3.0.17, 3.1.12, 3.2.24, 3.3.17, 3.4.7"
 
 # shellcheck disable=SC2039
-ETCD_NAME="${ETCD_NAME:-etcd-${ETCD_HOSTNAME:-$(hostname -s)}}"
+# NOTE: Keep ETCD_NAME in sync with host_name in https://github.com/kubernetes/kubernetes/blob/2e7996e3e2712684bc73f0dec0200d64eec7fe40/cluster/gce/gci/configure-helper.sh#L1563
+ETCD_NAME="${ETCD_NAME:-etcd-${ETCD_HOSTNAME:-$HOSTNAME}}"
```

Comment:
Hmm - this is no longer in sync now :)

Also:
1. Who is setting "HOSTNAME" ?
2. Why we have the "etcd-" prefix ?

Reply:
> Who is setting "HOSTNAME" ?
On the master vm, $HOSTNAME is indeed set to the master vm name. But my bad that I don't really know who set it, I got the `/usr/local/bin/migrate-if-needed.sh: line 49: hostname: command not found` when testing out, and realize @dims also changed hostname to HOSTNAME in #91171. Let me try find this out.

> Why we have the "etcd-" prefix
This is to agree with host name set in configure-helper.sh: https://github.com/kubernetes/kubernetes/blob/1bc526aa6c310b915e7ad57b4442deec7328aad2/cluster/gce/gci/configure-helper.sh#L1626

Reply:
@wenjiaswe right. the `hostname` binary is no longer available in the distroless etcd.

Reply:
Thanks! Do you know where the HOSTNAME is set? I knew it’s set cause I can see it on master, but I haven’t found who did it.

Reply:
Now that ETCD_HOSTNAME this seems less important, but since the default is to fallback to $HOSTNAME, might still be worth answering how it works for containerd?

Reply:
Checked with @Random-Liu , containerd does have  HOSTNAME set too: https://github.com/containerd/cri/blob/ae8200bf93e9eda91d1c8b40dc14257ec0bdea02/pkg/server/container_create_unix.go#L145.